### PR TITLE
nvc++ w/ OMP: work-around atomic capture

### PR DESCRIPTION
**Pull request by ax3l**
## Summary

Internal compiler error in this region with NVHPC 21.9 when compiling for OpenMP (host-only).

## Additional background

Nvidia bug report: 3390723.

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Src/Base/AMReX_FabArrayUtility.H [hunk 1]
```diff
@@ -1540,8 +1540,9 @@ indexFromValue (FabArray<FAB> const& mf, int comp, IntVect const& nghost,
                 bool old;
 // we should be able to test on _OPENMP < 201107 for capture (version 3.1)
 // but we must work around a bug in gcc < 4.9
-#if defined(AMREX_USE_OMP) && defined(_OPENMP) && _OPENMP < 201307 // OpenMP 4.0
-#pragma omp critical (amrex_indexfromvalue)
+// And, with NVHPC 21.9, we see an ICE with the atomic capture.
```
Comment (ax3l) [inline Src/Base/AMReX_FabArrayUtility.H]:
```suggestion
// And, with NVHPC 21.9, we see an ICE with the atomic capture (NV bug: #3390723)
```

File: Src/Base/AMReX_FabArrayUtility.H [hunk 2]
```diff
@@ -1540,8 +1540,9 @@ indexFromValue (FabArray<FAB> const& mf, int comp, IntVect const& nghost,
                 bool old;
 // we should be able to test on _OPENMP < 201107 for capture (version 3.1)
 // but we must work around a bug in gcc < 4.9
-#if defined(AMREX_USE_OMP) && defined(_OPENMP) && _OPENMP < 201307 // OpenMP 4.0
-#pragma omp critical (amrex_indexfromvalue)
+// And, with NVHPC 21.9, we see an ICE with the atomic capture (NV bug: #3390723)
+#if defined(AMREX_USE_OMP) && defined(_OPENMP) && (_OPENMP < 201307 || defined(__NVCOMPILER)) // OpenMP 4.0
+#pragma omp critical (amrex_indexfromvalueaa)
```
Comment (WeiqunZhang) [inline Src/Base/AMReX_FabArrayUtility.H]: Why the change `amrex_indexfromvalue` -> `amrex_indexfromvalueaa`?

Comment (ax3l) [reply Src/Base/AMReX_FabArrayUtility.H]: Oh, leftover from testing

Comment (ax3l) [reply Src/Base/AMReX_FabArrayUtility.H]:
```suggestion
#pragma omp critical (amrex_indexfromvalue)
```

**Comment by WeiqunZhang** [comment]
Looks like Intel has released oneapi 2021.4.0.  We have to fix the use of a deprecated function.

**Comment by ax3l** [comment]
Sounds good - @WeiqunZhang can you take that on in a separate PR pls? Currently occupied with MS.

**Comment by WeiqunZhang** [comment]
Yes, I will do it in a separate PR.  Looks like we are also having issues with HIP CI.  Will make them non-required temporarily.

**Review by WeiqunZhang — APPROVED** [review]

**Comment by ax3l** [comment]
Yes, HIP seems to have a temporary download issue in startups.
I tried restarting some, which helps in 2 out of 3 cases at the moment.
